[MRG+1] correct comparison in GaussianNB for 'priors'

gxyd opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/master/CONTRIBUTING.md#pull-request-checklist
-->

#### Reference Issues/PRs

Fixes #9633 
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->


#### What does this implement/fix? Explain your changes.

correct comparison in GaussianNB for 'priors'

#### Any other comments?

I am not fully sure if I have written the test as correctly since, I have written it as `clf.fit(X, Y)` and I don't know if I should use `assert_....` or not. That is why I have not put `MRG` on title of PR.

<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

gxyd committed:
fixes #9633 correct comparison in GaussianNB for 'priors'

jnothman commented on sklearn/tests/test_naive_bayes.py:
```diff
@@ -114,6 +114,16 @@ def test_gnb_priors():
+    priors = np.array([0.08, 0.14, 0.03, 0.16, 0.11, 0.16, 0.07, 0.14, 0.11, 0.0])
+    Y = np.array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10])
+    clf = GaussianNB(priors)
+    clf.fit(X, Y)
```
Comment "smoke test for issue #9633"

jnothman commented on sklearn/tests/test_naive_bayes.py:
```diff
@@ -114,6 +114,16 @@ def test_gnb_priors():
 
 
+def test_gnb_priors_sum_isclose():
+    """Test whether the class prior sum is properly tested"""
```
Use a comment rather than a docstring here. We do this because nose hides the function name when docstring is used.

jnothman reviewed (approved):
I've not yet double-checked that this test fails on master (does it?), but this otherwise lgtm

gxyd committed:
fix flake8 errors and address changes regarding comments

gxyd commented:
I have addressed the comments and fixed the failing flake8 errors. Do I need to squash the commits? (I have 2 commits now) Was wondering if you follow some strict policy against 'squashing' public commits.

jnothman commented:
what is the benefit of squashing commits at this stage? we squash when we
merge, and until then it's good to have a record of what's changed since
the last review, and to be able to follow links to commits.

On 26 Oct 2017 4:22 pm, "Gaurav Dhingra" <notifications@github.com> wrote:

> I have addressed the comments and fixed the failing flake8 errors. Do I
> need to squash the commits? (I have 2 commits now) Was wondering if you
> follow some strict policy against 'squashing' public commits.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/10005#issuecomment-339554618>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEz6zvxy9ZPyCvLYVqvOOo7MIOC0XqUks5swBcXgaJpZM4QGh4g>
> .
>

gxyd commented:
That is good to know, since I too was wondering as to why I don't see no 
'merge commits' in git history, your comment explains it.

TomDLT commented:
Squashed and merged, thanks @gxyd !

gxyd commented:
Thanks @TomDLT . This was my first contribution to scikit-learn.

TomDLT commented:
Congrats on your first contribution, hoping to see you around again !

gxyd commented:
I'll be here.

jnothman commented:
Forgot a changelog entry. @gxyd do you mind submitting a new PR with a change to doc/whats_new?

gxyd commented:
@jnothman see #10025 .
